modules/async_worker.py

```diff
@@ -838,6 +839,7 @@ def callback(step, x0, x, total_steps, y):
             task = async_tasks.pop(0)
             try:
                 running_task = task
+                script_callbacks.before_task_callback(task.task_id)
```

Comment:
Should you pass in the entire task, so it could be consumed by other scripts, instead of just passing in the task id? But we could extend it when neccessary